src/utils/service_builder.rs

```diff
@@ -0,0 +1,136 @@
+use crate::key_id_managers::{KeyIdManagerConfig, KeyIdManagerType, ManageKeyIDs};
+use crate::providers::{
+    core_provider::CoreProviderBuilder, mbed_provider::MbedProviderBuilder, Provide,
+};
+use parsec_interface::operations_protobuf::ProtobufConverter;
+use parsec_interface::requests::AuthType;
+use parsec_interface::requests::{BodyType, ProviderID};
+use serde::Deserialize;
+use std::path::PathBuf;
+use std::sync::Arc;
+use std::sync::RwLock;
+use std::time::Duration;
+use threadpool::{Builder as ThreadPoolBuilder, ThreadPool};
+
+const VERSION_MINOR: u8 = 0;
+const VERSION_MAJOR: u8 = 1;
+
+#[derive(Deserialize)]
+pub struct CoreSettings {
+    pub thread_pool_size: Option<usize>,
+
+    #[serde(default)]
+    pub idle_listener_sleep_duration: Option<u64>,
+}
+
+#[derive(Deserialize)]
+pub struct ServiceConfig {
+    pub core_settings: CoreSettings,
+    pub listener: ListenerConfig,
+    pub key_manager: KeyIdManagerConfig,
+}
+
+pub struct ServiceBuilder;
+
+impl ServiceBuilder {
+    pub fn build_service(config: &ServiceConfig) -> FrontEndHandler {
+        let on_disk_key_id_manager = get_key_id_manager(&config.key_manager);
+        let mbed_provider = MbedProviderBuilder::new()
+            .with_key_id_store(on_disk_key_id_manager)
+            .build();
+
+        // Store provider descriptions in it.
+        let core_provider = CoreProviderBuilder::new()
+            .with_version(VERSION_MINOR, VERSION_MAJOR)
+            .with_provider_info(mbed_provider.describe())
+            .build();
+
+        let mbed_backend_handler = BackEndHandlerBuilder::new()
+            .with_provider(Box::from(mbed_provider))
+            .with_converter(Box::from(ProtobufConverter {}))
+            .with_provider_id(ProviderID::MbedProvider)
+            .with_content_type(BodyType::Protobuf)
+            .with_accept_type(BodyType::Protobuf)
+            .with_version(VERSION_MINOR, VERSION_MAJOR)
+            .build();
+
+        let core_provider_backend = BackEndHandlerBuilder::new()
+            .with_provider(Box::from(core_provider))
+            .with_converter(Box::from(ProtobufConverter {}))
+            .with_provider_id(ProviderID::CoreProvider)
+            .with_content_type(BodyType::Protobuf)
+            .with_accept_type(BodyType::Protobuf)
+            .with_version(VERSION_MINOR, VERSION_MAJOR)
+            .build();
+
+        let dispatcher = DispatcherBuilder::new()
+            .with_backend(ProviderID::CoreProvider, core_provider_backend)
+            .with_backend(ProviderID::MbedProvider, mbed_backend_handler)
+            .build();
+
+        let simple_authenticator = Box::from(SimpleAuthenticator {});
+
+        FrontEndHandlerBuilder::new()
+            .with_dispatcher(dispatcher)
+            .with_authenticator(AuthType::Simple, simple_authenticator)
+            .build()
+    }
+
+    pub fn start_listener(config: &ListenerConfig) -> Box<dyn Listen> {
```

Comment:
Why not `build_listener` 😛 ? Would be nice just for coherency with the other functions.

Reply:
It felt more natural - you build a service, but you start listening (instead of build listening :P )